src/cpp/statistics/fastdds/domain/DomainParticipantImpl.cpp

```diff
@@ -134,7 +137,28 @@ ReturnCode_t DomainParticipantImpl::enable_statistics_datawriter(
                 return ReturnCode_t::RETCODE_ERROR;
             }
 
-            statistics_listener_->set_datawriter(event_kind, data_writer);
+            if (PHYSICAL_DATA_TOPIC == use_topic_name)
+            {
+                PhysicalData notification;
+                notification.participant_guid(*reinterpret_cast<const detail::GUID_s*>(&guid()));
+                std::string unique_id(reinterpret_cast<char*>(Host::instance().mac_id().value));
```

Comment:
Use `default_domain_id` from `QosPolicyUtils.hpp` instead of the direct cast, and then a a stringstream to convert the resulting `uint64_t` to string.